src/stores/search.ts

```diff
@@ -0,0 +1,341 @@
+    ) {
+      warn(
+        `Cannot toggle filter of type ${params.filterType}. Use code or codeIdx parameter`
+      )
+      return
+    }
+    const { filterType } = params
+    const codeIdx =
+      params.codeIdx ??
+      state.filters[filterType].findIndex((f) => f.code === params.code)
+    state.filters[filterType][codeIdx].checked =
+      !state.filters[filterType][codeIdx].checked
+  }
+
+  /**
+   * Resets all filters to initial values.
+   * Provider filters are not in the initial filters, so they need to be
+   * handled separately.
+   *
+   */
+  const clearFilters = () => replaceFilters(getBaseFiltersWithProviders())
+
+  /**
+   * Selecting some filter items disables related items. For example, selecting an `nc`
+   * license filter (CC BY-NC, CC BY-NC-SA, CC BY-NC-ND) disables the `Commercial` license
+   * type filter. This function determines if the filter item should be disabled based on
+   * the currently checked filter items.
+   */
+  function isFilterDisabled(
+    item: FilterItem,
+    filterCategory: FilterCategory
+  ): boolean | undefined {
+    if (!['licenseTypes', 'licenses'].includes(filterCategory)) {
+      return
+    }
+    if (item.code === 'commercial' || item.code === 'modification') {
+      const targetCode = {
+        commercial: 'nc',
+        modification: 'nd',
+      }[item.code]
+      return state.filters.licenses.some(
+        (item) => item.code.includes(targetCode) && item.checked
+      )
+    } else {
+      const dependentFilters: string[] = []
+      if (item.code.includes('nc')) {
+        dependentFilters.push('commercial')
+      }
+      if (item.code.includes('nd')) {
+        dependentFilters.push('modification')
+      }
+      return (
+        state.filters.licenseTypes.filter(
+          (item) => dependentFilters.includes(item.code) && item.checked
+        ).length > 0
+      )
+    }
+  }
+
+  /**
+   * Returns the object with filters for selected search type, with codes, names for i18n labels, and checked status.
+   */
+  const searchFilters: ComputedRef<Filters> = computed(() => {
+    return mediaFilterKeys[state.searchType]
+      .filter((filterKey) => filterKey !== 'mature')
+      .reduce((obj, filterKey) => {
+        obj[filterKey] = state.filters[filterKey]
+        return obj
+      }, {} as Filters)
+  })
+
+  /**
+   * Called when a /search path is server-rendered.
+   */
+  function setSearchStateFromUrl({
+    path,
+    urlQuery,
+  }: {
+    path: string
+    urlQuery: { [key: string]: string }
+  }) {
+    if (urlQuery.q) {
+      setSearchTerm(urlQuery.q.trim())
+    }
+    state.searchType = queryStringToSearchType(path)
+    // When setting filters from URL query, 'mature' has a value of 'true',
+    // but we need the 'mature' code. Creating a local shallow copy to prevent mutation.
+    const query = { ...urlQuery }
+    if (query.mature === 'true') {
+      query.mature = 'mature'
+    } else {
+      delete query.mature
+    }
```

Comment:
It seems like in the API if mature is set to anything at all that is truthy it gets evaluated as true, not just if it's explicitly `"true"`.

https://github.com/WordPress/openverse-api/blob/f2335988b1ef5d030f3e3bca9cadbed4a012ba9b/api/catalog/api/controllers/search_controller.py#L202-L205

Reply:
Nice! Thank you for the link.